### PR TITLE
Validate transaction call type

### LINKED ISSUE #250: Consolidate orgs and users as top-level entities
Following up on #245 and to complete radicle-dev/registry-spec#11 we need to consolidate orgs and uesrs and support the unified registration of projects under either as well as guarantee a shared namespace for their ids.

- [x] consolidate id namespace for orgs and users
- [ ] support project registration under both orgs and users https://github.com/radicle-dev/radicle-registry/issues/389

### LINKED ISSUE #269: Validate inherent extrinsics
At the moment nodes allow inherent calls in signed extrinsics. For example a user can submit a signed extrinsic with a `set_timestamp` call to a node. The node will accept that extrinsic, broadcast it, and it will eventually be included in a block. Executing that intrinsic will have no effect but it waste space, time, and bandwidth. The problem also applies to other inherents that we will add, like block rewards.

To address this we should add code that ensures that certain calls are only valid for unsigned extrinsics. This validation needs to be run by the tx pool and the block importer.

### PR DESCRIPTION
**Pull request by geigerzaehler**
To fix #269 and fix #250 we forbid certain extrinsic calls from entering the transaction pool and being executed.

### CODE REVIEW

**Comment by NunoAlexandre** [comment]
@geigerzaehler this approach looks good, although - before seeing code - I'd expect us to to do this using a signed extension where this would run at pre-dispatch. Did you think of that? Would it be a good idea, or advantageous?

**Comment by geigerzaehler** [comment]
> I'd expect us to to do this using a signed extension where this would run at pre-dispatch

I’ve considered this but it requires substantially more machinery and is less transparent. We would need to create an empty `struct` just for the purpose of implementing `SignedExtension`. We would need to add that struct to `SignedExtra` and also add it to `transaction_extra_to_runtime_extra` in the client. Moreover `SignedExtension` forces us to separate validation of signed and unsigned extrinsics which gives us less flexibility in structuring the code.

Also note that hardcoding the `InvalidTransaction` values in this manner is not ideal. This is something to pick up on when we tackle the errors.

File: runtime/src/lib.rs [hunk 1]
```diff
@@ -243,6 +245,36 @@ pub type Executive = frame_executive::Executive<
     AllModules,
 >;
 
+const SIGNED_INHERENT_CALL_ERROR: InvalidTransaction = InvalidTransaction::Custom(1);
+const FOBIDDEN_CALL_ERROR: InvalidTransaction = InvalidTransaction::Custom(2);
+const UNSGINED_CALL_ERROR: InvalidTransaction = InvalidTransaction::Custom(3);
+
+/// Validate that the call of extrinsic is allowed.
```
Comment (NunoAlexandre) [inline runtime/src/lib.rs]:
```suggestion
/// Validate that the call of the extrinsic is allowed.
```

Comment (NunoAlexandre) [reply runtime/src/lib.rs]: or

```suggestion
/// Validate that the extrinsic call is allowed.
```

File: runtime/src/lib.rs [hunk 2]
```diff
@@ -243,6 +245,36 @@ pub type Executive = frame_executive::Executive<
...
+///
+/// * We forbid calls reserved for inherents when the extrinsic is not signed.
+/// * We forbid any calls to the [Balances] or [System] module.
+/// * We ensure that the extrinsic is signed for non-inherent calls.
+///
+fn validate_extrinsic_call(xt: &UncheckedExtrinsic) -> Result<(), InvalidTransaction> {
+    match xt.function {
+        // Inherents are only allowed if there unsigned.
```
Comment (NunoAlexandre) [inline runtime/src/lib.rs]:
```suggestion
        // Inherents are only allowed if they are unsigned.
```

**Pull request merged**